### PR TITLE
feat(components): Combine label with form input components

### PR DESCRIPTION
**Pull request by justman00**
Addresses https://github.com/sumup-oss/circuit-ui/issues/569

## Purpose

To simplify the Inputs API by integrating the label inside together

## Approach and changes

For labels and inputs there is obviously 2 approached, one is wrapping the input inside the label tag so that they are implicitly one for another or use the **htmlFor** property in order to do it explicitly. Unfortunately the first option is a no-go because we want to support also the **visuallyHidden** property.

The PR also fixes some of the issues found on the fly when developing in storybook and specifically the issues in the Select [component](https://circuit.sumup.com/?path=/story/forms-select--with-prefix).

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/components/Input/Input.js [hunk 1]
```diff
@@ -438,7 +449,19 @@ Input.propTypes = {
...
+  /**
+   * A string that will become the label for the input.
+   */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  /**
+   * Defines whether the label should be hidden or not.
+   */
+  labelVisuallyHidden: PropTypes.bool,
+  /**
+   * Sets the label for the input, needed for a11y.
```
Comment (connor-baer) [inline src/components/Input/Input.js:462]:
```suggestion
   * A unique identifier for the input field. If not defined, a randomly generated id is used.
```

File: src/components/Select/Select.js [hunk 2]
```diff
@@ -311,7 +321,19 @@ Select.propTypes = {
...
+   */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  /**
+   * Defines whether the label should be hidden or not.
+   */
+  labelVisuallyHidden: PropTypes.bool,
+  /**
+   * Sets the label for the input, needed for a11y.
+   */
+  id: PropTypes.string.isRequired
```
Comment (connor-baer) [inline src/components/Select/Select.js:336]: Same suggestion as for the Input component :)

**Comment by connor-baer** [comment]
> The PR also fixes some of the issues found on the fly when developing in Storybook and specifically the issues in the Select component.

What are the issues that you're referring to here?

**Comment by justman00** [comment]
> > The PR also fixes some of the issues found on the fly when developing in Storybook and specifically the issues in the Select component.
> 
> What are the issues that you're referring to here?

Yes so https://circuit.sumup.com/?path=/story/forms-select--with-prefix has no prefix although it should have and not any of the examples are interactive, also if you open the console there is an overflow of warnings. Thank you so much for the feedback @connor-baer , will work on addressing it 🙏

**Comment by connor-baer** [comment]
The issues with the Select stories were already fixed in the `alpha` branch AFAIK 😉

**Comment by justman00** [comment]
> The issues with the Select stories were already fixed in the `alpha` branch AFAIK 😉
> 
> Let me know if you have any questions or wanna pair program.

That's awesome, I had no idea that they were fixed haha. Yes thank you will keep you updated

File: src/components/Select/Select.js (hunk 2 above)
Comment (justman00) [reply src/components/Select/Select.js]: updated

File: src/components/Input/Input.js [hunk 3]
```diff
@@ -295,39 +298,47 @@ const StyledInput = ({
...
-        disabled,
-        className: wrapperClassName,
-        css: wrapperStyles
-      }}
-    >
-      {prefix}
-      <InputElement
+    <>
+      {label ? (
+        <Label htmlFor={props.id} visuallyHidden={labelVisuallyHidden}>
```
[the remark this answers is not in the document]
Comment (justman00) [reply src/components/Input/Input.js]: updated

File: src/components/Select/Select.js [hunk 4]
```diff
@@ -322,7 +344,10 @@ Select.defaultProps = {
   placeholder: 'Select an option',
   inline: false,
   noMargin: false,
-  renderPrefix: null
+  renderPrefix: null,
+  label: '',
+  labelVisuallyHidden: false,
+  id: undefined
```
[the remark this answers is not in the document]
Comment (justman00) [reply src/components/Select/Select.js]: updated

File: src/components/AutoCompleteInput/AutoCompleteInput.story.js [hunk 5]
```diff
@@ -58,14 +57,14 @@ const icons = {
 const AutoCompleteInputWithLabel = props => {
   const id = uniqueId();
   return (
-    <Label htmlFor={id}>
-      {"What's your favourite fruit?"}
+    <div>
```
[the remark this answers is not in the document]
Comment (justman00) [reply src/components/AutoCompleteInput/AutoCompleteInput.story.js]: updated

**Pull request merged**